src/main/java/com/wanderersoftherift/wotr/rift/objective/definition/CollectObjective.java

```diff
@@ -0,0 +1,91 @@
+package com.wanderersoftherift.wotr.rift.objective.definition;
+
+import com.mojang.serialization.MapCodec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import com.wanderersoftherift.wotr.core.rift.RiftConfig;
+import com.wanderersoftherift.wotr.rift.objective.ObjectiveType;
+import com.wanderersoftherift.wotr.rift.objective.OngoingObjective;
+import com.wanderersoftherift.wotr.rift.objective.ongoing.CollectOngoingObjective;
+import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.core.registries.Registries;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.util.ExtraCodecs;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.level.ServerLevelAccessor;
+import net.minecraft.world.level.storage.loot.LootTable;
+
+import java.util.Optional;
+
+public class CollectObjective implements ObjectiveType {
+
+    public static final MapCodec<CollectObjective> CODEC = RecordCodecBuilder.mapCodec(inst -> inst.group(
+            ExtraCodecs.POSITIVE_INT.fieldOf("min_quantity").forGetter(CollectObjective::getMinQuantity),
+            ExtraCodecs.POSITIVE_INT.fieldOf("max_quantity").forGetter(CollectObjective::getMaxQuantity),
+
+            BuiltInRegistries.ITEM.byNameCodec()
+                    .optionalFieldOf("item")
+                    .forGetter(o -> Optional.ofNullable(o.getItem())),
+
+            ResourceLocation.CODEC.optionalFieldOf("loot_table")
```

Comment:
Can use `ResourceKey.codec(Registries.LOOT_TABLE)` here, and skip the need to convert the resource location.